worker/src/RTC/TransportCongestionControlClient.cpp

```diff
@@ -244,35 +290,52 @@ namespace RTC
 		// more stable values.
 		this->bitrates.startBitrate = std::max<uint32_t>(MinBitrate, this->bitrates.availableBitrate);
 
+		auto prevMaxBitrate = this->bitrates.maxBitrate;
+
 		if (this->desiredBitrateTrend.GetValue() > 0u)
 		{
 			this->bitrates.maxBitrate = std::max<uint32_t>(
 			  this->initialAvailableBitrate,
 			  this->desiredBitrateTrend.GetValue() * MaxBitrateIncrementFactor);
 
-			if (this->maxOutgoingBitrate > 0u)
+			// If max bitrate requested didn't change by more than a small % keep the previous settings
+			// to avoid constant small fluctuations requiring extra probing and making the estimation
+			// less stable (requires constant redistribution of bitrate accross consumers).
+			auto maxBitrateMargin = this->bitrates.maxBitrate * MaxBitrateMarginFactor;
+			if (
+			  prevMaxBitrate > this->bitrates.maxBitrate - maxBitrateMargin &&
+			  prevMaxBitrate < this->bitrates.maxBitrate + maxBitrateMargin)
 			{
-				this->bitrates.maxBitrate =
-				  std::min<uint32_t>(this->maxOutgoingBitrate, this->bitrates.maxBitrate);
+				this->bitrates.maxBitrate = prevMaxBitrate;
 			}
-
-			this->bitrates.maxPaddingBitrate = this->bitrates.maxBitrate * MaxPaddingBitrateFactor;
 		}
 		else
 		{
-			this->bitrates.maxBitrate        = this->initialAvailableBitrate;
-			this->bitrates.maxPaddingBitrate = 0u;
+			this->bitrates.maxBitrate = this->initialAvailableBitrate;
 		}
 
+		if (this->maxOutgoingBitrate > 0u)
+		{
+			this->bitrates.maxBitrate =
+			  std::min<uint32_t>(this->maxOutgoingBitrate, this->bitrates.maxBitrate);
+		}
+		this->bitrates.maxPaddingBitrate = this->bitrates.maxBitrate * MaxPaddingBitrateFactor;
```

Comment:
These also don't need to run unless something has changed. Very little work, but it is often wasteful.

Reply:
Yes, it makes sense to me skipping operations if there is no need to do them.

Can we please @ggarber make the function return if there is nothing new to tell to `rtpTransportControllerSend`?